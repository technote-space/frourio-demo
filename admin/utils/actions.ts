import type { PageKeys } from '~/_pages';
import type { Dispatch } from '~/store';

export const setAdmin = (dispatch: Dispatch, admin: { name?: string; icon?: string }) => dispatch({
  type: 'SET_ADMIN',
  admin,
});

export const openSidebar = (dispatch: Dispatch) => dispatch({ type: 'OPEN_SIDEBAR' });
export const closeSidebar = (dispatch: Dispatch) => dispatch({ type: 'CLOSE_SIDEBAR' });
export const openLicense = (dispatch: Dispatch) => dispatch({ type: 'OPEN_LICENSE' });
export const closeLicense = (dispatch: Dispatch) => dispatch({ type: 'CLOSE_LICENSE' });

export const changePage = (dispatch: Dispatch, page: PageKeys) => dispatch({ type: 'PAGE', page });
export const changeTitle = (dispatch: Dispatch, title: string) => dispatch({ type: 'TITLE', title });
export const logout = (dispatch: Dispatch) => dispatch({ type: 'LOGOUT' });

export const setNotice = (dispatch: Dispatch, message: string): void => dispatch({
  type: 'SET_NOTICE',
  notice: { message },
});
export const setError = (dispatch: Dispatch, message: string): void => dispatch({
  type: 'SET_ERROR',
  notice: { message },
});
export const setWarning = (dispatch: Dispatch, message: string): void => dispatch({
  type: 'SET_WARNING',
  notice: { message },
});
export const closeNotice = (dispatch: Dispatch): void => dispatch({ type: 'CLOSE_NOTICE' });
