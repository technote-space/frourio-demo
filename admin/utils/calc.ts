import { differenceInCalendarDays } from 'date-fns';

export const getPriceCalc = (price: number, number: number, checkin: string | Date, checkout: string | Date, amount: number): string => {
  const diff = differenceInCalendarDays(new Date(checkout), new Date(checkin));
  if (diff <= 0) {
    return '';
  }

  const calculated = price * number * diff;
  return (amount === calculated ? '' : `¥${calculated} = `) + `¥${price} * ${number}人 * ${diff}泊`;
};
