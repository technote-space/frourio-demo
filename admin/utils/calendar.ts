import type { EventApi } from '@fullcalendar/react';
import { eachDayOfInterval, subDays, format } from 'date-fns';
import FullCalendar from '@fullcalendar/react';

export const getEventDates = (calendar: FullCalendar | null): Array<string> =>
  [...new Set((calendar?.getApi().getEvents() ?? []).filter(
    (event): event is EventApi & { start: Date; end: Date; } => !!event.start && !!event.end,
  ).flatMap(event => eachDayOfInterval({
    start: event.start,
    end: subDays(event.end, 1),
  })).map(date => format(date, 'yyyy-MM-dd')))].sort();
