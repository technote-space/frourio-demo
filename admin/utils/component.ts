import type { FC } from 'react';

export const getDisplayName = <T extends FC = FC>(prefix: string, Component: T) => `${prefix}(${Component.displayName || Component.name || 'Component'})`;

export const addDisplayName = <T extends FC = FC, U extends FC = FC>(prefix: string, Component: T, WrappedComponent?: U) => {
  Component.displayName = getDisplayName(prefix, WrappedComponent ?? Component);
  return Component;
};
