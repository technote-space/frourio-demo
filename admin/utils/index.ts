import { singular, plural } from 'pluralize';

export const getWord = (word: string, number: number): string => number <= 1 ? singular(word) : plural(word);
