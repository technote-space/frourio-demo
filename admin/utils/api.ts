/* eslint-disable @typescript-eslint/no-explicit-any */
import type { AspidaResponse } from 'aspida';
import type { ApiInstance } from '$/api/$api';
import type { MaybeUndefined } from '~/types';
import type { Dispatch } from '~/store';
import aspida from '@aspida/axios';
import { AxiosError } from 'axios';
import { singular } from 'pluralize';
import api from '$/api/$api';
import { logout, setError, onRefreshToken } from '~/utils/actions';

const apiClient = api(aspida());
export const client = {
  ...apiClient.admin,
  login: apiClient.login.admin,
};

export const isAxiosError = (target: any): target is AxiosError => {
  return typeof target === 'object' &&
    'config' in target &&
    'request' in target &&
    'response' in target;
};

export const handleAuthError = async <T, U, V, API extends (...args: Array<any>) => Promise<AspidaResponse<T, U, V>>>(
  dispatch: Dispatch,
  fallback: T,
  api: API,
  ...option: Parameters<API>
): Promise<T | MaybeUndefined<T>> | never => {
  try {
    const result = await api(...option);
    return result.body;
  } catch (error) {
    console.log(error);
    /* istanbul ignore next */
    if (isAxiosError(error)) {
      console.log(error.response);
      if (error.response && 401 === error.response.status) {
        if (error.response.data?.tokenExpired) {
          setError(dispatch, error.response.data?.message);
          logout(dispatch);
        } else {
          setError(dispatch, error.response.data?.message ?? error.response.config.method?.toUpperCase() === 'GET' ? '' : 'その操作をする権限がありません。');
          onRefreshToken(dispatch);
        }

        if (!fallback) {
          throw error;
        }

        return fallback;
      }

      /* istanbul ignore next */
      if (error.response) {
        setError(dispatch, error.response.data?.message);
        throw error;
      }
    }

    /* istanbul ignore next */
    setError(dispatch, error.message);
    /* istanbul ignore next */
    throw error;
  }
};

export type DataTableApi = {
  get: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  post: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  detail: (id: number) => {
    patch: (option?: any) => Promise<AspidaResponse<any, any, any>>;
    delete: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  }
};
export type DataTableApiModels = {
  [key in keyof ApiInstance['admin']]: ApiInstance['admin'][key] extends {
    get,
    post,
  } ? key : never
}[keyof ApiInstance['admin']]
export const getDataTableApi = (model: DataTableApiModels): DataTableApi => ({
  ...client[model],
  detail: id => ({
    patch: client[model][`_${singular(model)}Id`](id).patch,
    delete: client[model][`_${singular(model)}Id`](id).delete,
  }),
});

export const processUpdateData = <T extends Partial<{
  updatedAt,
  createdAt,
}>>(data: T): Omit<T, 'updatedAt' | 'createdAt'> => {
  delete data.updatedAt;
  delete data.createdAt;
  return data;
};
