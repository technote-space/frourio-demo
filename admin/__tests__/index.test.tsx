import dotenv from 'dotenv';
import Index from '~/pages/index';
import { render, setup, useNock, setToken, setInvalidToken, setDarkMode, loadPage, act } from '~/__tests__/utils';
import user from '@testing-library/user-event';

dotenv.config({ path: '../server/.env' });

setup();

describe('Index', () => {
  it('should show login form', async() => {
    const { asFragment, findByText, getAllByText, getByText, getByLabelText } = render(<Index/>);

    await findByText('Login');
    expect(getAllByText('予約システム')).toHaveLength(2); // header, footer
    expect(getByText('Email address')).toBeVisible();
    expect(getByLabelText(/Email address/)).toBeVisible();
    expect(getByText('Password')).toBeVisible();
    expect(getByLabelText(/Password/)).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fail to login', async() => {
    const login = jest.fn();
    useNock()
      .post('/login/admin', body => {
        login(body);
        return body;
      }).reply(400);
    setInvalidToken();

    const { getByText, getByLabelText, getByTestId, findByText } = render(<Index/>);

    await findByText('Login');
    user.type(getByLabelText(/Email address/), 'test@example.com');
    user.type(getByLabelText(/Password/), 'password');
    user.click(getByTestId('password-switch'));
    await act(async() => {
      user.click(getByText('Login'));
    });

    await findByText('Request failed with status code 400');

    expect(login).toBeCalledWith({ email: 'test@example.com', pass: 'password' });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fail to login (invalid header)', async() => {
    const login = jest.fn();
    useNock()
      .post('/login/admin', body => {
        login(body);
        return body;
      }).reply(204);

    const { getByText, getByLabelText, findByText } = render(<Index/>);

    await findByText('Login');
    user.type(getByLabelText(/Email address/), 'test@example.com');
    user.type(getByLabelText(/Password/), 'password');
    await act(async() => {
      user.click(getByText('Login'));
    });

    await findByText('Invalid header');

    expect(login).toBeCalledWith({ email: 'test@example.com', pass: 'password' });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should success to login', async() => {
    const login = jest.fn();
    useNock()
      .post('/login/admin', body => {
        login(body);
        return body;
      }).reply(204, undefined, {
        authorization: 'test-token',
      })
      .get('/admin').reply(200, {
        name: null,
        icon: 'http://localhost:8080/api/icons/dummy.svg',
        roles: [
          { 'role': 'dashboard', 'name': 'Dashboard' },
        ],
      })
      .get('/admin/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);

    const { getByText, getByLabelText, findByText, findAllByText, getAllByText, container, findByTestId } = render(
      <Index/>,
    );

    await findByText('Login');
    user.type(getByLabelText(/Email address/), 'test@example.com');
    user.type(getByLabelText(/Password/), 'password');
    await act(async() => {
      user.click(getByText('Login'));
    });

    expect(await findAllByText('チェックイン')).toHaveLength(2); // table title, table header

    expect(login).toBeCalledWith({ email: 'test@example.com', pass: 'password' });
    expect(getAllByText('チェックアウト')).toHaveLength(2); // table title, table header
    expect(getByText('全部屋')).toBeVisible();

    // test header
    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    expect(buttons).toHaveLength(2);
    user.click(getAllByText('予約システム')[0]); // header link
    user.click(buttons[1]); // toggle dark mode

    // test sidebar
    user.click(buttons[0]);
    await findByTestId('admin-avatar');
    const menuClose = await findByTestId('drawer-layer');
    user.click(menuClose);
  });

  it('should logout automatically', async() => {
    useNock()
      .get('/admin').reply(401, {
        message: 'test error',
        tokenExpired: true,
      });
    setToken('token');

    const { findByText } = render(<Index/>);

    await findByText('Login');
  });

  it('should logout automatically if no appropriate roles', async() => {
    useNock().get('/admin').reply(200, {
      name: 'test name', icon: null, roles: [],
    });
    setToken('token');

    const { findByText } = render(<Index/>);

    await findByText('Login');
  });

  it('should not logout automatically', async() => {
    useNock()
      .get('/admin').reply(200, {
        name: 'test name',
        icon: null,
        roles: [
          { 'role': 'dashboard', 'name': 'Dashboard' },
        ],
      })
      .get('/admin/dashboard/rooms').reply(401)
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);
    setToken('token');

    const { findByTestId } = render(<Index/>);

    await findByTestId('select-date');
  });

  it('should show fallback page', async() => {
    useNock()
      .get('/admin').reply(200, {
        name: 'test name', icon: null, roles: [
          { 'role': 'rooms', 'name': 'Rooms' },
        ],
      })
      .get(/rooms\?/)
      .reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      });
    setToken('token');

    const { findByTestId } = render(<Index/>);

    await findByTestId('page-rooms');
  });

  it('should login automatically and logout', async() => {
    useNock()
      .get('/admin').reply(200, {
        name: 'test name',
        icon: null,
        roles: [
          { 'role': 'dashboard', 'name': 'Dashboard' },
        ],
      })
      .get('/admin/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);
    setToken('token');
    setDarkMode(true);

    const { getByText, findByText, findByTestId, container } = render(<Index/>);

    await findByTestId('select-date');

    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    user.click(buttons[0]);
    await findByText('test name');
    user.click(getByText('ログアウト'));
    await findByText('Login');
    expect(getByText('Email address')).toBeVisible();
    expect(getByText('Password')).toBeVisible();
  });

  it('should show license dialog', async() => {
    useNock()
      .get('/admin').reply(200, {
        name: 'test name',
        icon: null,
        roles: [
          { 'role': 'dashboard', 'name': 'Dashboard' },
        ],
      })
      .get('/admin/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);
    setToken('token');

    const { findByText, findByTestId, findAllByTestId, container } = render(<Index/>);

    await findByTestId('select-date');

    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    user.click(buttons[0]);
    user.click(await findByText('ライセンス'));
    user.click((await findAllByTestId('license-item'))[0]);
    user.click(await findByTestId('close-license'));
    user.click(await findByTestId('close-license-list'));
  });

  it('should show license dialog without login', async() => {
    const { findByText, findByTestId, findAllByTestId, container } = render(<Index/>);

    await findByText('Login');

    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    user.click(buttons[0]);
    user.click(await findByText('ライセンス'));
    user.click((await findAllByTestId('license-item'))[0]);
    user.click(await findByTestId('close-license'));
    user.click(await findByTestId('close-license-list'));
  });

  it('should reuse component', async() => {
    const { findByText, findByTestId, container } = await loadPage(
      'guests',
      scope => scope
        .get(/guests\?/).reply(200, {
          'data': [],
          'page': 0,
          'totalCount': 0,
        }),
    );

    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    user.click(buttons[0]);
    user.click(await findByText('ダッシュボード'));
    await findByTestId('select-date');
  });
});
