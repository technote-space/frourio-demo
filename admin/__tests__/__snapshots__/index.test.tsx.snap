// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should fail to login (invalid header) 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter makeStyles-content-69"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiSnackbarContent-root makeStyles-warning-101 makeStyles-content-69 MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-104"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-102 makeStyles-iconVariant-103"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          Invalid header
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-102"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-wrap-77"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-80 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-81 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-82 MuiTypography-h6"
        >
          予約システム
        </h6>
        <div
          class="makeStyles-grow-83"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main
      class="makeStyles-main-78"
    >
      <div
        class="makeStyles-contents-79"
      >
        <div
          class="makeStyles-wrap-90"
          data-testid="page-dashboard"
        >
          <div
            class="makeStyles-wrap-91"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-94"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value="test@example.com"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-94"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-95"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-96 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="password"
                          value="password"
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-97"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          data-testid="password-switch"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Show
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="makeStyles-login-92"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-88"
    >
      <div
        class="makeStyles-wrap-89"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Index should fail to login 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter makeStyles-content-32"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiSnackbarContent-root makeStyles-warning-64 makeStyles-content-32 MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-67"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-65 makeStyles-iconVariant-66"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          Request failed with status code 400
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-65"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-wrap-40"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-43 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-44 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-45 MuiTypography-h6"
        >
          予約システム
        </h6>
        <div
          class="makeStyles-grow-46"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main
      class="makeStyles-main-41"
    >
      <div
        class="makeStyles-contents-42"
      >
        <div
          class="makeStyles-wrap-53"
          data-testid="page-dashboard"
        >
          <div
            class="makeStyles-wrap-54"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-57"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value="test@example.com"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-57"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-58"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-59 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="text"
                          value="password"
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-60"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          data-testid="password-switch"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Hide
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="makeStyles-login-55"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-51"
    >
      <div
        class="makeStyles-wrap-52"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Index should show login form 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrap-10"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-13 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-14 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h6
          class="MuiTypography-root makeStyles-title-15 MuiTypography-h6"
        >
          予約システム
        </h6>
        <div
          class="makeStyles-grow-16"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main
      class="makeStyles-main-11"
    >
      <div
        class="makeStyles-contents-12"
      >
        <div
          class="makeStyles-wrap-23"
          data-testid="page-dashboard"
        >
          <div
            class="makeStyles-wrap-24"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-27"
                  >
                    <label
                      class="MuiFormLabel-root Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-27"
                  >
                    <label
                      class="MuiFormLabel-root Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-28"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-29 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-30"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          data-testid="password-switch"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Show
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="makeStyles-login-25"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-21"
    >
      <div
        class="makeStyles-wrap-22"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
